pageserver/src/bin/pageserver.rs

```diff
@@ -337,9 +348,16 @@ fn start_pageserver(conf: &'static PageServerConf) -> Result<()> {
             page_service::thread_main(conf, auth, pageserver_listener, conf.auth_type)
         })?;
 
-    page_service_thread
-        .join()
-        .expect("Page service thread has panicked")?;
+    join_handles.push(page_service_thread);
 
+    // Initialize tenant manager.
+    tenant_mgr::init(conf);
```

Comment:
Why move this? Also, was it originally in the right place? Should the initialization happen before any of the worker threads that would touch `REPOSITORY`

Reply:
I think I was shuffling the code quite intensively and it ended up here. Agree with you, though it should work just fine because accesses to repo occur only during connection handling and by that time initialization completes. But yeah, it is better not to leave a place for errors there